impeller/renderer/backend/vulkan/deletion_queue_vk.cc

```diff
@@ -0,0 +1,27 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/renderer/backend/vulkan/deletion_queue_vk.h"
+
+namespace impeller {
+
+DeletionQueueVK::DeletionQueueVK() = default;
+
+DeletionQueueVK::~DeletionQueueVK() {
+  Flush();
+}
+
+void DeletionQueueVK::Flush() {
+  for (auto it = deletors_.rbegin(); it != deletors_.rend(); ++it) {
+    (*it)();
+  }
```

Comment:
Is this always going to be used from a single thread?

Are we sure that deletors won't enqueue additional things to delete when they're deleted?

We should document threading expectations in the header.